pyproject.toml

```diff
@@ -66,7 +66,7 @@ mkdocs-jupyter = ">=0.19.0"
 pymdown-extensions = ">=9.2"
 pygments = ">=2.11.1"
 responses = ">=0.14"
-ruff = "^0.1.6"
+ruff = "^0.4.2"
```

Comment:
Why only ruff got bumped?

Reply:
I suspect that's because it's the only dependency using `^` instead of `>=`, but more importantly, why did mkdocs fail?